pkg/state/types.go

```diff
@@ -1602,3 +1602,41 @@ func (v1 *Vault) EqualWithOpts(v2 *Vault, ignoreID, ignoreTS bool) bool {
 	}
 	return reflect.DeepEqual(v1Copy, v2Copy)
 }
+
+type License struct {
+	kong.License `yaml:",inline"`
+	Meta
+}
+
+// TODO: add a variable definition to notate that License
+// (and definition of other entities) should satisfy crud.Event interface?
+
+func (l *License) Identifier() string {
+	return *l.ID
+}
+
+func (l *License) Console() string {
+	return l.FriendlyName()
+}
+
+func (l *License) Equal(l2 *License) bool {
+	return l.EqualWithOpts(l2, false, false)
+}
+
+func (l *License) EqualWithOpts(l2 *License, ignoreID, ignoreTS bool) bool {
+	l1Copy := l.License.DeepCopy()
+	l2Copy := l2.License.DeepCopy()
+
+	if ignoreID {
+		l1Copy.ID = nil
+		l2Copy.ID = nil
+	}
+	if ignoreTS {
+		l1Copy.CreatedAt = nil
+		l2Copy.CreatedAt = nil
+
+		l1Copy.UpdatedAt = nil
+		l2Copy.UpdatedAt = nil
+	}
+	return reflect.DeepEqual(l1Copy, l2Copy)
+}
```

Comment:
The `License` type and all the methods on it are missing godocs. The other types and functions in this file do have those so I'd suggest we are consistent and add those as well here.

Reply:
done.